src/PositionDescriptor.sol

```diff
@@ -0,0 +1,146 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.26;
+
+import {IPoolManager} from "@uniswap/v4-core/src/interfaces/IPoolManager.sol";
+import {Currency, CurrencyLibrary} from "@uniswap/v4-core/src/types/Currency.sol";
+import {IHooks} from "@uniswap/v4-core/src/interfaces/IHooks.sol";
+import {StateLibrary} from "@uniswap/v4-core/src/libraries/StateLibrary.sol";
+import {PoolKey} from "@uniswap/v4-core/src/types/PoolKey.sol";
+import {PoolId, PoolIdLibrary} from "@uniswap/v4-core/src/types/PoolId.sol";
+import {IERC20Metadata} from "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
+import {IPositionManager} from "./interfaces/IPositionManager.sol";
+import {IPositionDescriptor} from "./interfaces/IPositionDescriptor.sol";
+import {PositionInfo, PositionInfoLibrary} from "./libraries/PositionInfoLibrary.sol";
+import {Descriptor} from "./libraries/Descriptor.sol";
+import {CurrencyRatioSortOrder} from "./libraries/CurrencyRatioSortOrder.sol";
+import {SafeERC20Namer} from "./libraries/SafeERC20Namer.sol";
+
+/// @title Describes NFT token positions
+/// @notice Produces a string containing the data URI for a JSON metadata string
+contract PositionDescriptor is IPositionDescriptor {
+    using StateLibrary for IPoolManager;
+    using PoolIdLibrary for PoolKey;
+    using CurrencyLibrary for Currency;
+    using PositionInfoLibrary for PositionInfo;
+
+    // mainnet addresses
+    address private constant DAI = 0x6B175474E89094C44Da98b954EedeAC495271d0F;
+    address private constant USDC = 0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48;
+    address private constant USDT = 0xdAC17F958D2ee523a2206206994597C13D831ec7;
+    address private constant TBTC = 0x8dAEBADE922dF735c38C80C7eBD708Af50815fAa;
+    address private constant WBTC = 0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599;
+
+    address public immutable WETH9;
+    /// @dev A null-terminated string
+    bytes32 public immutable nativeCurrencyLabelBytes;
+
+    IPoolManager public immutable poolManager;
+
+    constructor(IPoolManager _poolManager, address _WETH9, bytes32 _nativeCurrencyLabel) {
+        poolManager = _poolManager;
+        WETH9 = _WETH9;
+        nativeCurrencyLabelBytes = _nativeCurrencyLabel;
+    }
+
+    /// @notice Takes the native currency label in bytes32 and returns it as a string
+    /// @return nativeCurrencyLabel The native currency label
+    function nativeCurrencyLabel() public view returns (string memory) {
```

Comment:
Whats the purpose of this function? Why do we actually need to process a bytes32 into a string?